### DISCUSSION TITLE
Query reguring rdkit bond type

### DISCUSSION THREAD

**Discussion by shrimonmuke0202** [rdkit/rdkit#5359] [Q&A]
Dear Team,
In Rdkit, there are 22 bond types present. But in smile strings, four bond types mainly exist (i.e., single, double, triple, and aromatic). But I want to incorporate the remaining bond types (like dative, quadruple, hydrogen, ionic, quintuple, and hextuple) in smiles notation. It will be highly helpful if you kindly share your thoughts regarding the above matter and any other extra information is always welcome. Waiting for your positive reply.

**Comment by greglandrum** [comment]
The RDKit actually supports two SMILES extensions which allow quadruple bonds (`$`) and dative bonds (`->`) already:
```
In [4]: m = Chem.MolFromSmiles("O->[Os]$[Os]<-O")

In [5]: m.Debug()
Atoms:
	0 8 O chg: 0  deg: 1 exp: 0 imp: 2 hyb: 5 arom?: 0 chi: 0
	1 76 Os chg: 0  deg: 2 exp: 5 imp: 0 hyb: 3 arom?: 0 chi: 0
	2 76 Os chg: 0  deg: 2 exp: 5 imp: 0 hyb: 3 arom?: 0 chi: 0
	3 8 O chg: 0  deg: 1 exp: 0 imp: 2 hyb: 5 arom?: 0 chi: 0
Bonds:
	0 0->1 order: 17 conj?: 0 aromatic?: 0
	1 1->2 order: 4 conj?: 0 aromatic?: 0
	2 3->2 order: 17 conj?: 0 aromatic?: 0

In [6]: Chem.MolToSmiles(m)
Out[6]: 'O->[Os]$[Os]<-O'
```

We don't currently have extensions to read/write the other bond types as SMILES or any plans to add them. There are only so many characters that one can use in SMILES and I think we may be there already.
